carsus/io/cmfgen/util.py

```diff
@@ -0,0 +1,332 @@
+import gzip
+import itertools
+
+import astropy.constants as const
+import astropy.units as u
+import numpy as np
+
+RYD_TO_EV = u.rydberg.to('eV')
+H_IN_EV_SECONDS = const.h.to('eV s').value
+HC_IN_EV_ANGSTROM = (const.h * const.c).to('eV angstrom').value
+
+
+def open_cmfgen_file(fname, encoding='ISO-8859-1'):
+    return gzip.open(fname, 'rt') if fname.endswith('.gz') else open(fname, encoding=encoding) 
+
+
+def to_float(string):
+    """
+    String to float, also deals with Fortran 'D' type.
+
+    Parameters
+    ----------
+    string : str
+
+    Returns
+    -------
+    float
+    """
+    try:
+        value = float(string.replace('D', 'E'))
+
+    except ValueError:
+
+        # Typo in `MG/VIII/23oct02/phot_sm_3000`, line 23340
+        if string == '1-.00':
+            value = 10.00
+
+        # Typo in `SUL/V/08jul99/phot_op.big`, lines 9255-9257
+        if string == '*********':
+            value = np.nan
+
+    return value
+
+
+def find_row(fname, string1, string2=None, how='AND', row_number=False):
+    """
+    Search for strings in plain text files and returns the matching\
+    line (or row number).
+
+    Parameters
+    ----------
+    fname : str
+        Path to plain text file.
+    string1 : str
+        String to search.
+    string2 : str
+        Secondary string to search (default is None).
+    how : {'OR', 'AND', 'AND NOT'}
+        Search method: `string1` <method> `string2`
+            (default is 'AND').
+    row_number : bool
+        If true, returns row number (default is False).
+
+    Returns
+    -------
+    str or int
+        Returns matching line or match row number.
+    """
+
+    if string2 is None:
+        string2 = ''
+
+    with open_cmfgen_file(fname) as f:
+        n = 0
+        for line in f:
+
+            n += 1
+            if how == 'OR':
+                if string1 in line or string2 in line:
+                    break
+
+            if how == 'AND':
+                if string1 in line and string2 in line:
+                    break
+
+            if how == 'AND NOT':
+                if string1 in line and string2 not in line:
+                    break
+
+        else:
+            n, line = None, None
+
+    if row_number is True:
+        return n
+
+    return line
```

Comment:
```suggestion
    return line
```
returning strings or ints?